### ISSUE TITLE
User configurable option for temporal resolution on tracks

### ISSUE THREAD

**Issue by zubenubi** [SignalK/freeboard-sk#113]
When using the option to display the vessel track stored in influxdb, the default temporal resolution is as follows:

- the last 1hr -> resolution=5s`
- the previous 23h -> resolution=1m
- prior to that -> resolution=5m

In some cases this default setting causes the displayed track to lose detailed points in the interest of improving responsiveness of the app on low powered devices. 

A user-configurable option for this setting would be desirable so that end users can tune the setting to their own needs. If possible the temporal resolution would be able to be specified from 1s to 300s, with the default remaining at 5s. Currently the setting for retrieving points for the track allows a timeframe to be specified 0h-96h, so users will be able to readily observe if the rendering of track is too costly and be able to adjust accordingly.

**Comment by tkurki** [comment]
Another option would be to derive the temporal resolution from the scale of the map. Just age does not make a detailed track less interesting, but if you are zoomed out more trackpoints don't give any value.

Could be based on map zoom level or actual map viewport size.

**Comment by zubenubi** [comment]
I agree. Scale-dependent rendering would be a nice feature. At the moment the temporal filter causes the track to sometimes cross land areas, so an option to be able to see the most detailed track in freeboard

See attached
________________________________

**Comment by panaaj** [comment]
Implementing temporal filtering on the client will require the transfer of data to that will be ultimately discarded.
Is there an opportunity to apply the temporal filtering on the server to transfer only the resultant points?

If filtering is done by the client and 96hrs of track is required to be displayed, for the filtering to provide an acceptable result you would request track points at a minimum of 5sec resolution for the last 60min (720) and 30sec resolution for the remaining 95hrs (11400)... most of which will be discarded. 

Just a thought.

**Comment by zubenubi** [comment]
Is the performance likely to suffer a lot if higher temporal resolutions are made available for configuration? If so maybe it would be possible to alert the end user or prevent the track from being high resolution for anything > 24 hours ?

**Comment by panaaj** [comment]
@zubenubi I have done some work on this and have landed on the following operational parameters:
```
the last 1 hour:  5 sec track resoution
the next 23 hrs:  10 sec track resolution
beyond 24 hrs:  1 minute track resolution
```
Based on my testing 3000 points of track is simplified to approx 600 pts with no discernable visual difference.
I will look to make a pre-release available.

**Comment by panaaj** [comment]
@zubenubi I have released a test version of Freeboard-SK v1.22.0 which has the ability to define the resolution of the track retrieved from the server.
You will need to install it  on the SignalK server from the command line.

```
cd ~/.signalk
npm i @signalk/freeboard-sk@uat
```
Version should be 1.22.0

**Comment by panaaj** [comment]
Published v 1.22.1 with server track  simplification
